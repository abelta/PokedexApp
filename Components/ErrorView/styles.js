import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  errorView: {
    padding: 20,
  },
  text: {
    fontSize: 20,
    textAlign: 'center',
  },
});

export default styles;
