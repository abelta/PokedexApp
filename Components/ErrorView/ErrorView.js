import React from 'react';
import PropTypes from 'prop-types';
import { Text, View } from 'react-native';
import styles from './styles';

const ErrorView = ({ children, style }) => (
  <View style={[styles.errorView, style]}>
    <Text style={styles.text}>{children}</Text>
  </View>
);

ErrorView.propTypes = {
  children: PropTypes.node.isRequired,
  style: PropTypes.number, // eslint-disable-line react/require-default-props
};

export default ErrorView;
