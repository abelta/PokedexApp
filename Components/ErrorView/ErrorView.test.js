import React from 'react';
import renderer from 'react-test-renderer';
import ErrorView from '.';

describe('ErrorView', () => {
  const tree = renderer.create(<ErrorView>Some text</ErrorView>).toJSON();
  it('renders correctly', () => {
    expect(tree).toMatchSnapshot();
  });
});
