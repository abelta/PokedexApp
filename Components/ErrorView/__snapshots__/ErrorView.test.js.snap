// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorView renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "padding": 20,
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 20,
        "textAlign": "center",
      }
    }
  >
    Some text
  </Text>
</View>
`;
